api/turing/it/database/database.go

```diff
@@ -76,7 +59,15 @@ func CreateTestDatabase() (*gorm.DB, func(), error) {
 		}
 	}
 
-	if testDb, err = migrate(testDb, testDbName); err != nil {
+	dbCfg := &config.DatabaseConfig{
```

Comment:
you can actually initialize an instance of `config.DatabaseConfig` at `api/turing/it/database/config.go` as the user-provided input is being read there

Reply:
I thought about it but the database always changes with a new test suite, so the object will have to be created upon invocation of `WithTestDatabase()`

Reply:
well, your global variables are guaranteed to be initialized before `WithTestDatabase` is invoked. I.e. your variables here: https://github.com/gojek/turing/blob/0e8eec8042bbb074240b9aa7d4f4fb2303a3ca9c/api/turing/it/database/config.go#L7-L13
are guaranteed to be initialized, so why can't you simply replace it with:
```go
var dbConfig = config.DatabaseConfig {
	Host:              getEnvOrDefault("DATABASE_HOST", "localhost"),
	Port:              strconv.Atoi(getEnvOrDefault("DATABASE_PORT", "5432")),
	User:              getEnvOrDefault("DATABASE_USER", "turing"),
	Password:          getEnvOrDefault("DATABASE_PASSWORD", "turing"),
	Database:          getEnvOrDefault("DATABASE_NAME", "turing"),
	MigrationsFolder:  "../db-migrations/",
}
```

Reply:
Ya ok i did something like this but mutate as needed with a helper method.

Reply:
The current state is quite confusing as 2 kinds of db are being juggled so i refactored it for readability as well.